Detectors/TRD/calibration/include/TRDCalibration/T0Fit.h

```diff
@@ -0,0 +1,111 @@
+// or submit itself to any jurisdiction.
+
+/// \file   T0Fit.h
+/// \brief  Fits the TRD PH distributions to extract the t0 value
+/// \author Luisa Bergmann
+
+#ifndef O2_TRD_T0FIT_H
+#define O2_TRD_T0FIT_H
+
+#include "DataFormatsTRD/T0FitHistos.h"
+#include "DetectorsCalibration/TimeSlotCalibration.h"
+#include "DetectorsCalibration/TimeSlot.h"
+#include "DataFormatsTRD/Constants.h"
+#include "CCDB/CcdbObjectInfo.h"
+#include "DataFormatsTRD/CalT0.h"
+#include "TRDCalibration/CalibrationParams.h"
+
+#include "Rtypes.h"
+#include "TH2.h"
+#include "TH1.h"
+#include "TProfile.h"
+#include "TF1.h"
+#include "Fit/Fitter.h"
+#include "TFile.h"
+#include "TTree.h"
+
+#include <array>
+#include <cstdlib>
+#include <memory>
+
+namespace o2
+{
+namespace trd
+{
+//______________________________________________________________________________________________
+struct ErfLandauChi2Functor {
+  double operator()(const double* par) const;
+  std::vector<float> x; ///< x-value (time-bin) of adc profile
+  std::vector<float> y; ///< y-value (av. adc) for corresp. time-bin
+  float lowerBoundFit;  ///< lower bound for fit
+  float upperBoundFit;  ///< upper bound for fit
+};
+
+//______________________________________________________________________________________________
+class T0Fit final : public o2::calibration::TimeSlotCalibration<o2::trd::T0FitHistos>
+{
+  using Slot = o2::calibration::TimeSlot<o2::trd::T0FitHistos>;
+
+ public:
+  T0Fit() = default;
+  ~T0Fit() final = default;
+
+  bool hasEnoughData(const Slot& slot) const final { return slot.getContainer()->getNEntries() >= mMinEntriesTotal; }
+  void initOutput() final;
+  void finalizeSlot(Slot& slot) final;
+  Slot& emplaceNewSlot(bool front, TFType tStart, TFType tEnd) final;
+
+  /// (Re-)Creates a file "trd_t0fit.root". This lets continually fill
+  /// a tree with the fit results.
+  void createOutputFile();
+
+  /// Close the output file. E.g. First write the tree to the file and let the
+  /// smart pointers take care of closing the file.
+  void closeOutputFile();
+
+  const std::vector<o2::trd::CalT0>& getCcdbObjectVector() const { return mObjectVector; }
+  std::vector<o2::ccdb::CcdbObjectInfo>& getCcdbObjectInfoVector() { return mInfoVector; }
+
+  void initProcessing();
+
+  /// Initialize the fit values once with the previous valid ones if they are
+  /// available.
+  void retrievePrev(o2::framework::ProcessingContext& pc);
+
+ private:
+  bool mInitDone{false};                                         ///< flag to avoid creating output etc multiple times
+  const TRDCalibParams& mParams{TRDCalibParams::Instance()};     ///< reference to calibration parameters
+  size_t mMinEntriesTotal{mParams.minEntriesTotalT0Fit};         ///< minimum total number of angular deviations
+  size_t mMinEntriesChamber{mParams.minEntriesChamberT0Fit};     ///< minimum number of angular deviations per chamber for accepting refitted value
```

Comment:
why are you not using directly `mParams.minEntriesChamberT0Fit` instead of creating another member to the class?

Reply:
no real reason, other than I did the same for the total number of entries. Should I change mMinEntriesTotal as well, or leave that as is for readability (in hasEnoughData())?

Reply:
I'd do the same also for the other variable

Reply:
replaced both variables with the direct instance of the calibration parameters